lib/cocoapods/command/spec.rb

```diff
@@ -37,9 +37,12 @@ def validate_regex!(query)
       #         Whether the paths for all the versions should be returned or
       #         only the one for the last version.
       #
+      # @param  [String] version
+      #         Optional version to search for in the specs.
+      #
       # @return [Pathname] the absolute path or paths of the given podspec
       #
-      def get_path_of_spec(spec, show_all = false)
+      def get_path_of_spec(spec, show_all = false, version = nil)
```

Comment:
This odd thing with this is that `show_all` and `version` are exclusive. If `version` is specified, `show_all` will be ignored, and vice-versa. Especially it's not clear just from the method signature and doc what would happen with `show_all = true, version = '1.2.3'` 

I see 2 approaches to solve this:
 - Either clearly document which one takes precedence. In other words, in the documentation for `show_all`, add a mention that "this parameter is ignored if a non-nil `version` is provided"
 - Or combine the two parameters into one, which imho is the better and more idiomatic to ruby. Since Ruby is dynamically typed, it's totally valid to allow a parameter to be either a bool or a String. This would give something like this:

```ruby
# @param. [String] spec
#         The name of the specification.
#
# @param  [Bool,String] version_filter
#         - If set to false, will return only the spec path for the latest version (the default).
#         - If set to true, will return a list of all paths of all the versions of that spec.
#         - If set to a String, will return only the spec path for the version specified by that string.
# 
# @return [Pathname, Array<Pathname>] the absolute path or paths of the given podspec
#
def get_path_of_spec(spec, version_filter)
  …
  if version_filter.is_a?(String)
    all_paths_from_set(set, version_filter).split(/\n/).first
  elsif version_filter == true
    all_paths_from_set(set) # all
  else # latest
    …
  end
end
```

Reply:
[Bonus] To be completely honest, an even nicer API (and more idiomatic of ruby) would be:
 - To have `version_filter` be either a `String` or a `Symbol` (rather than "`String` or `Bool`), and use the symbol `:all` instead of value `true` to mean "all versions", and the symbol `:latest` instead of value `false` to mean "latest version only".
 - I also wonder why `all_paths_from_set` returns a multi-line string that the caller (here `get_path_of_spec`) needs to `.split(/\n/)`, instead of have `all_paths_from_set` be the one returning an `Array<String>` itself

But both those changes, however nice they would be by making the API of this method way more idiomatic, would require updating _all_ the call sites as well, which means a broader change and larger PR. So if you don't feel up for this, I can totally understand, and that's why the intermediate solution above (have `version_filter` be a "`String` or `Bool`" and not a "`String` or `Symbol`") would still be fine, as it would still keep retro-compatibility with existing code without needing to change anything in the rest of the codebase.

Reply:
Makes total sense, I had it like you suggested but then it was a huge mess and I reverted half of it and used the two arguments. Now that I have a clearer picture of what I am doing I chose the `version_filter` option.

I converted `all_paths_from_set` to return an array and modified the callers.